src/frequenz/sdk/timeseries/_ringbuffer/buffer.py

```diff
@@ -108,6 +108,17 @@ def gaps(self) -> List[Gap]:
         """
         return self._gaps
 
+    def is_missing_value(self, value: QuantityT | None) -> bool:
```

Comment:
Nitpick:
```suggestion
    def is_value_missing(self, value: QuantityT | None) -> bool:
```

Or simply `is_missing()`. If you take a `Sample` instead of a `QuantityT`, I think `is_value_missing()` makes more sense.

Reply:
Also, using a negative as part of a function name is usually not recommended, because you can end up with a double negation, which is harder to reason about. What about:

```suggestion
    def has_value(self, sample: Sample) -> bool:
```

But I'm not 100% sure in this case, because I guess here *missing* is important, as you are talking about gaps. Another alternative could be `is_value_present(sample)`. So just food for thought, I'm not convinced either way :P

Also, infinite is considered as not missing, right?

Reply:
I can make it taking a sample instead. Do you think it can be moved to a method of `Sample` at some point?

Will change it to `has_value`. There is already a `is_missing`, which is about timestamps. I guess another candidate to improve naming.

Reply:
> Also, infinite is considered as not missing, right?

So far yes, on the typical microgrid data we work with infinities will already be turned into NaNs and for other use-cases I did not want to restrict it too much.

Reply:
> I can make it taking a sample instead. Do you think it can be moved to a method of Sample at some point?

I'm not sure, because again what's a present value could depend on the application, for example is `infinite` a value or not? I would be in favor of adding more common methods from `math` that would also check for `None`, like `is_finite()`, but that is `not isnan() and not isinf()`.

> There is already a `is_missing`, which is about timestamps.  I guess another candidate to improve naming.

Oh! Yes :)

Reply:
Another alternative could be `is_gap()` which ties it up pretty well with the concept.

But I was thinking if maybe `is_missing(timestamp)` can be unified into a `is_gap(sample)`, then `update()` could already pass if a timestamp is missing to `self._update_gaps()`. But maybe this is too much of a refactor to be on the scope of this PR.

Reply:
Not sure about `is_gap(sample)`: `is_missing` checks whether a sample is already in our gap list, `has_value` checks whether a sample can be considered a gap.